### PR TITLE
Avoid `can_cast` calling to reduce overhead

### PR DESCRIPTION
**Pull request by okuta**
`numpy.can_cast` is very slow. rel #2673.

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
Thank you for your advice! I will fix.

**Comment by okuta** [comment]
jenkins, test this please.

File: cupy/core/_kernel.pyx [hunk 1]
```diff
@@ -334,6 +334,16 @@ cdef tuple _broadcast(list args, tuple params, bint use_size):
     return value, shape
 
 
+cdef _numpy_can_cast = numpy.can_cast
+
+
+cdef bint _can_cast(d1, d2, casting):
+    # almost ufunc passes `same_kind`
```
Comment (toslunar) [inline cupy/core/_kernel.pyx]:
```suggestion
    # most ufunc passes `same_kind`
```

**Comment by okuta** [comment]
Thanks. I fixed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**